### PR TITLE
BAAS-26308: Add StringifyError to Exception struct

### PR DESCRIPTION
**Pull request by tcannon91**

### CODE REVIEW

File: runtime_test.go [hunk 1]
```diff
@@ -2971,6 +2971,76 @@ func TestExceptionWithinAppliedObjectFunc(t *testing.T) {
...
+		{
+			description: "Object",
+			exceptionVal: func() Value {
+				testObject := runtime.NewObject()
+				testObject.Set("foo", "bar")
+				testObject.Set("fizz", "buzz")
+
+				return testObject
+			},
+			expectedString: "{\"foo\":\"bar\",\"fizz\":\"buzz\"}",
```
Comment (UnicodeSnowman) [inline runtime_test.go]: [q] are goja objects always ordered? just want to make sure the keys in this string will always be in the same order.

**Review by UnicodeSnowman — APPROVED** [review]
👏

File: runtime_test.go (hunk 1 above)
Comment (tcannon91) [reply runtime_test.go]: I am looking to see if I can figure this out definitively. Anecdotally, I ran the tests with `-count=30` and did not see any changes in the order.

Comment (UnicodeSnowman) [reply runtime_test.go]: good enough for me

Comment (tcannon91) [reply runtime_test.go]: It uses a go map under the hood, whose iteration order is not guaranteed. I think I will try to update this. Probably just going to break it into a separate test and get a little clever with the string comparison.

Comment (tcannon91) [reply runtime_test.go]: Got a new test up there, let me know what you think

**Review by UnicodeSnowman — APPROVED** [review]
solid, LGTM

**Pull request merged**